devicetypes/smartthings/zigbee-accessory-dimmer.src/zigbee-accessory-dimmer.groovy

```diff
@@ -115,10 +115,16 @@ def on() {
 def setLevel(value, rate = null) {
 	if (value == 0) {
 		sendEvent(name: "switch", value: "off")
+		// OneApp expects a level event when the dimmer value is changed
+		value = device.currentValue("level")
 	} else {
 		sendEvent(name: "switch", value: "on")
-		sendEvent(name: "level", value: value)
 	}
+	runIn(1, delayedSend, [data: createEvent(name: "level", value: value), overwrite: true])
```

Comment:
Remind me why can't the event be sent right away?

Reply:
The "setLevel" command makes it up to the cloud _after_ the level event otherwise, so OneApp doesn't associate them and satisfy its own response requirements.